radix-engine-interface/src/model/resource/non_fungible_id.rs

```diff
@@ -8,26 +8,89 @@ use sbor::*;
 
 use crate::abi::*;
 use crate::data::*;
+use crate::math::Decimal;
 use crate::scrypto_type;
+use crate::Describe;
 
 /// Represents a key for a non-fungible resource
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct NonFungibleId(pub Vec<u8>);
+pub struct NonFungibleId {
```

Comment:
As Josh mentioned, it's better to refactor `NonFungibleId` into an array, and do the byte conversion at encode/decode time.